### ISSUE TITLE
Opening Admin UI locks DB for production site using JS SDK

### ISSUE THREAD

**Issue by zettadam** [pocketbase/pocketbase#5749]
When I open built-in Admin UI in a browser tab, it usually takes about 60 seconds for the UI to appear.

During that time, the web app I created and host in production (opened in another browser tab) that connects to DB file is prevented from reading DB and returning results until admin UI finishes loading (which I find unusually slow).

I know that with SQLite there can only be one write at a time, but why are reads (using JS SDK) blocked when Admin UI is loading? Why is this happening?

**Comment by ganigeorgiev** [comment]
Please don't open issues like this. I consider it low effort and lazy.

Without any details about your setup, what amount of data you are operating with, are there are any custom hooks code, transactions, indexes, etc. I really don't know how to help you.

In any case, for general purpose questions consider opening a [Q&A discussion](https://github.com/pocketbase/pocketbase/discussions/new/choose) with more details about your environment and ideally some steps to reproduce. See also https://github.com/pocketbase/pocketbase/discussions/3859#discussioncomment-7744045.

**Comment by zettadam** [comment]
When you say it is "low effort and lazy"... what do you mean? I followed the documentation to construct queries with filters API. I use stock Pocketbase executable on my server with defaults applied. I don't see anything else in the docs that would inform a user of such issues.

When I open Admin UI, it opens `users` table by default which has 1 record. (That takes about 60 seconds to load). My DB has miniscule amount of data (between 100-200 records in total spread between 6 tables, one of them has a relation to another table).

When the Admin UI eventually finishes loading, my web app is extemely fast (it loads results--and I usually have 2 queries per "page"--in less than a 1s).

From a user point of view, loading Admin UI is a affecting other queries and it is an "issue". What other information might be helpful to investigate this issue? Perhaps I could post it here.

**Comment by ganigeorgiev** [comment]
By "low effort and lazy" I mean that there is literally no information how to reproduce it.

It literally reads like "It doesn't work on my computer. How to fix it?".

How to fix what exactly?

I'm not ruling out a PocketBase issue but your description just doesn't really make sense and there is nothing concrete to analyze or work on.

I'm using PocketBase in production for [Presentator.io](https://presentator.io/) and I'm not experiencing this (see more details about its setup in https://github.com/pocketbase/pocketbase/discussions/4254).

Additionally no other user so far has reported such issue.

Please understand that this is a non-commercial open source project I work on it on volunteer basis. I know that this may sound rude but if you expect dedicated support you'll be better using something else.

If you still want help, please:
- open a [Q&A discussion](https://github.com/pocketbase/pocketbase/discussions/new/choose)
- provide clear reproducible steps so me or someone could follow and attempt to reproduce your issue locally
- what are the versions of the PocketBase executable and JS SDK
- any additional information about your setup (eg. where it is deployed, are you behind a reverse proxy, etc.)

**Comment by zettadam** [comment]
Why did you close this issue? I'm beginning to think I probably made the wrong choice using Pocketbase.

**Comment by ganigeorgiev** [comment]
Because I don't think it is PocketBase related. The issue being closed indicates that no action can be further taken.

And again, without steps to reproduce I don't understand what is the expected behavior here.

I'll stop engage further. If you want help provide clear reproducible steps and I'll try to have a look at it but until then there is nothing much I can assist you here.

**Comment by zettadam** [comment]
I am using Pocketbase v0.22.21.

Here are the steps.

1. Open Pocketbase Admin UI in a browser tab (takes 30-60 seconds)
2. While Admin UI is loading, open web application using Pocketbase client SDK (JS) in another tab.
3. Wait until Admin UI finshes loading to see web application data appear.
4. When opening web app without opening Admin UI at the same time, the app queries take less that 1s.

Attached is are my collections.
[pb_schema.json](https://github.com/user-attachments/files/17572104/pb_schema.json)

I host Admin UI using nginx reverse proxy setup described in the docs but I don't think this should have any bearing on SQlite query performance in another hosted application.

**Comment by ganigeorgiev** [comment]
I've imported your schema locally but I'm still not able to reproduce your issue.

The steps I've taken:

- Initialized a brand new project with the prebuilt PocketBase v0.22.21 executable (no custom hooks or anything extra).
- Imported your db schema.
- Created 10 users.
- Created 100 `codes` and 100 `feeds` dummy records.
- Started apache bench several times in the background to send list requests to the 2 collections (e.g. `ab -n 100 -c 10 "http://127.0.0.1:8090/api/collections/{collection}/records"`)
- During the `ab` run I've tried to interact with the Admin UI and everything operated as expected

I don't how to assist you further.

My recommends are:
- Try to reproduce the issue locally with just PocketBase to rule out environment misconfiguration or 3rd-party issue (nginx could have an impact if not configured properly, e.g. enabled throttling, rate limiter, etc.)
- If you have custom `pb_hooks` remove them and test again
- Start the executable with `--dev` to inspect the actual query times

You can rule out directly step 1) by testing with the live demo - https://pocketbase.io/ (it uses a newer PocketBase v0.23.0-rc version but the record apis are the same).

Note that while PocketBase uses SQLite in WAL mode and read requests can executes concurrently, there is no guarantee that the operations will be processed in parallel/simultaneously. This depends on many other factors (e.g. available threads) and for all I know what you are experiencing could be even caused by throttling on CPU/network level by the hosting provider.

Again, I'm not ruling out PocketBase issue but v0.22.x is public for more than six months and no one so far has reported similar issue. As mentioned earlier I'm using PocketBase in production (see https://github.com/pocketbase/pocketbase/discussions/4254) and everything seems to work fine for me so far (the only slow requests I encounter based on the logs are the occasional thumbs generation due to the nature of the application).

If you still think that it is a PocketBase issue, please try to create a **minimal reproducible repo** with a sample `pb_data` that illustrates the problem and I'll to investigate it but I think I've spent more enough time based on the current provided steps.

---

Regarding _"I'm beginning to think I probably made the wrong choice using Pocketbase"_.

I'll repeat what I wrote earlier - this is a non-commercial/hobby open source project and I work on it on volunteer basis and **neither you or anyone else is entitled to anything more than what is already published**. I made no promises. I'm not here to sell you anything. I don't even benefit anything from you using PocketBase. If you think that I or somebody else owns you something then you free to use something else.

**Comment by zettadam** [comment]
Thank you for replying with the details. I'll follow your suggestions and will test it locally.

**Comment by zettadam** [comment]
One thing to keep in mind is that everything works very fast, even when I interact with Admin UI in production (once the initial load is finished). The initial load of Admin UI is what causes query delays in another app connected to the same DB--the rest of my app (JS, CSS and other files load as quickly as expected)

**Comment by ganigeorgiev** [comment]
Again, I'm not experiencing the described behavior. It is possible to be something specific related to your stored data, because I noticed that your schema has a lot of `editor` and `json` fields which often get misused (e.g. storing large base64 images), I don't really know.

You can try to experiment with different payloads and narrow down when the issue occur.

You can also try to open the browser's inspect tool and check the Network tab for anything suspicious like slow response time, slow JSON loading/parsing, etc. With v0.23.0-rc there are several extra protections and more reasonable defaults in place to limit and prevent resources misuse and I will consider eventually porting some of them later in v0.22.x but again, without clear reproducible steps the issue as it is described above doesn't really make sense to me.

I'll stop engage in pointless back-and-forth. Feel free to share a minimal reproducible repo and details about your setup once you've identified the culprit and I'll review it.

**Comment by zettadam** [comment]
"I'll stop engage in pointless back-and-forth" ... this is what bothers me about your attitude. By highlighting problems I or others might be having, even though you may think that it's not a problem, you have consistently been discouraging users from engaging with and potentially helping your product. Things you say to your users like "lazy and low effort", "pointless back-and-forth" might be indicative of something; I'll let you think about what exactly. I'm not entitled to anything more than it is already published, that's true, but I spent time to write this to help both myself and potentially Pocketbase.

**Comment by ganigeorgiev** [comment]
Fine, this will  be my last reply because I really don't want to argue with strangers on the internet, but will try to elaborate why I behave like this.

**Reporting an issue with a vague description and no clear details how to reproduce doesn't help.**

To understand this, you can try to imagine a standard work environment where a QA or PM assign a ticket to you with a very vague description of a problem. How would you proceed in that situation - you'll probably try to guess what the other party mean or will return the ticket back to the issuer to provide more details, right? At least that's what I'm doing.

Because of the nature of the project and the tight database integration every issue could be a potential security flow or even worst cause accidental data loss.

And I take security and bug reports very seriously.
When I receive a notification for an issue I literally stop working on everything else and do my best to investigate it and fix it as soon as possible.

Your issue will closed intermediately if I don't think it is PocketBase related and I'm not able reproduce it based on my understanding of the provided description.

If someone else has a similar issue they can simply open a new one. If I notice a pattern then probably there is an issue with PocketBase or maybe both users have applied the same misconfiguration.

See for example just from couple hours ago in https://github.com/pocketbase/pocketbase/discussions/5752. The issue described there is not PocketBase related. The provided initial reproducible is not complete and it is missing the actual Dockerfile the user is building with (the reproducible is missing the `encryptionEnv` configuration which the user clarified later as a reply because it ended up similar to another issue).

**In all cases, a closed issue is not the end of the world.** If you provide more details and proper reproducible steps then I'll reopen and reinvestigate it.

This has happened several times in the past, the most recent example probably is https://github.com/pocketbase/pocketbase/issues/5674. But the common part between those reopened issues is that often the initial description has very little or no relation to the actual issue, which results in wasted time and energy for both sides.

And I hope you can at least a little understand why I'm so direct, and maybe sometime annoyed/rude, when I ask for reproducible steps and such are not provided.
